### PR TITLE
Python 3.11 compatibility fix

### LINKED ISSUE #6: Dependencies not compatible with Python 3.11
Python 3.11 is not compatible with the version of matplotlib used in the current version of mvmlib.

### PR DESCRIPTION
**Pull request by johnmartins**
Some dependencies needed to be updated for mvmlib to be compatible with Python 3.11. This push request patches those dependencies, and updates the github workflow to work with the new dependencies. 

Dependency updates of note:
- Matplotlib
- Pandoc

Closes #6

### CODE REVIEW

**Comment by johnmartins** [comment]
Tests are successful when run locally, but workflow fails to build environment.

**Comment by johnmartins** [comment]
Now the tests complete. However, I had to update pandoc to work with Python 3.11. The new pandoc seems to want another input.

**Comment by johnmartins** [comment]
@khbalhandawi I had some trouble getting the workflow to work with Python 3.11. That's why there are so many commits (I suggest squashing them into a single commit if you do decide to merge this into your master). Since Pandoc needed to be updated to work with 3.11 there may be visual changes to the documentation. I've never used pandoc, so I'm not sure how to check for this.

File: tests/test_fuzzy.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import pytest
 import numpy as np
 import matplotlib.pyplot as plt
+import random
+
+random.seed(0)
```
Comment (khbalhandawi) [inline tests/test_fuzzy.py:6]: made these tests reproducible

File: tests/test_marginNetwork.py [hunk 2]
```diff
@@ -5,7 +5,9 @@
 import matplotlib
 from typing import List, Tuple
 from scipy.optimize import fsolve, minimize
+import random
 
+random.seed(0)
```
Comment (khbalhandawi) [inline tests/test_marginNetwork.py:10]: made these tests reproducible

File: tests/test_probFunction.py [hunk 3]
```diff
@@ -1,6 +1,9 @@
 import pytest
 import numpy as np
 import matplotlib.pyplot as plt
+import random
+
+random.seed(0)
```
Comment (khbalhandawi) [inline tests/test_probFunction.py:6]: made these tests reproducible

File: .github/workflows/push.yml
Comment (khbalhandawi) [inline .github/workflows/push.yml:1]: reduced duplicate workflow files
only two workflows, one for releases and one for pushing changes

File: .github/workflows/push.yml [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+    - name: Run unit tests
+      run: python -m pytest --import-mode=append tests/
+
+  docs:
+    name: documentation
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ '3.8', '3.9', '3.10', '3.11' ]
```
Comment (khbalhandawi) [inline .github/workflows/push.yml:52]: Add testing on multiple python versions to ensure backwards compatibility

File: .github/workflows/push.yml [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+  documentation-build-directory: ./docs/build
+  deploy-python-version: '3.11'
+
+jobs:
+  tests:
+    name: tests_master
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ '3.8', '3.9', '3.10', '3.11' ]
```
Comment (khbalhandawi) [inline .github/workflows/push.yml:16]: Add testing on multiple python versions to ensure backwards compatibility

File: .github/workflows/push.yml [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        python -m pip install --upgrade pip
+        python -m pip install -r requirements_dev.txt
+
+    - name: Build flit package and install dependencies
+      run: |
+        # install package using flit locally to run documentation notebooks
+        flit build
+        flit install --deps=develop --symlink
+
+    - name: Build documentation
```
Comment (khbalhandawi) [inline .github/workflows/push.yml:83]: Separate doc build from deployment (see deploy job below)

File: .github/workflows/push.yml [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+    - name: Archive built documentation
+      uses: actions/upload-artifact@v3
+      if: github.ref != 'refs/heads/master' # only on other branches
+      with:
+        name: documentation html (${{ matrix.python-version }})
+        path: ${{env.documentation-build-directory}}
+        retention-days: 1
+
+  deploy_docs:
```
Comment (khbalhandawi) [inline .github/workflows/push.yml:103]: Get build html files from cache and deploys to github pages only if on the master branch
Also creates and artifact if user wishes to download html docs locally and deploy manually

**Review by khbalhandawi — APPROVED** [review]
Squashed python 3.11 commits and then modifed workflow to test for python version 3.8 - 3.11

**Pull request merged**